src/map/script.c

```diff
@@ -14547,11 +14564,18 @@ BUILDIN(npctalk)
 		nd = map->id2nd(st->oid);
 	}
 
+	if (script_hasdata(st, 4)) {
+		show_npcname = script_getnum(st, 4);
+	}
+
 	if (nd != NULL) {
 		char name[NAME_LENGTH], message[256];
 		safestrncpy(name, nd->name, sizeof(name));
 		strtok(name, "#"); // discard extra name identifier if present
-		safesnprintf(message, sizeof(message), "%s : %s", name, str);
+		switch (show_npcname) {
```

Comment:
Yes please. The default case feels unnecessary if you're accepting boolean input here (anything != 0 is true).
Please also remove any mention of 1 and 0 in the documentation for this, and only keep true and false.